e2e/e2e_test.go

```diff
@@ -0,0 +1,2610 @@
+	}
+
+	// Test volume removal
+	// Remove the ConfigMap volume
+	if err := newCmd(t, "config", "volumes", "remove",
+		"--mount-path=/etc/config").Run(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Update implementation to verify ConfigMap is no longer accessible
+	impl = `package function
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+)
+
+func Handle(w http.ResponseWriter, _ *http.Request) {
+	errors := []string{}
+
+	// Check ConfigMap volume should NOT exist
+	if _, err := os.Stat("/etc/config"); !os.IsNotExist(err) {
+		errors = append(errors, "ConfigMap volume still exists after removal")
+	}
+
+	// Check Secret volume should still exist
+	secretData, err := os.ReadFile("/etc/secret/secret.txt")
+	if err != nil {
+		errors = append(errors, fmt.Sprintf("Secret read error: %v", err))
+	} else if string(secretData) != "secret-data" {
+		errors = append(errors, fmt.Sprintf("Secret data mismatch: got %q", string(secretData)))
+	}
+
+	// Check EmptyDir volume should still exist
+	testFile := "/tmp/scratch/test2.txt"
+	if err := os.WriteFile(testFile, []byte("test2"), 0644); err != nil {
+		errors = append(errors, fmt.Sprintf("EmptyDir write error: %v", err))
+	}
+
+	if len(errors) > 0 {
+		http.Error(w, strings.Join(errors, "\n"), http.StatusInternalServerError)
+		return
+	}
+	fmt.Fprintln(w, "OK")
+}
+`
+	err = os.WriteFile(filepath.Join(root, "handle.go"), []byte(impl), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Redeploy and verify removal worked
+	if err := newCmd(t, "deploy").Run(); err != nil {
+		t.Fatal(err)
+	}
+
+	if !waitForContent(t, fmt.Sprintf("http://%s.default.localtest.me", name), "OK") {
+		t.Fatalf("function failed after volume removal")
+	}
+}
+
+// TODO: TestMetadata_Subscriptions ensures that function instances can be
+// subscribed to events.
+func TestMetadata_Subscriptions(t *testing.T) {
+	// TODO
+	// Create a function which emits an event with as much defaults as possible
+	// Create a function which subscribes to those events
+	// Succeed the test as soon as it receives the event
+	t.Skip("Subscritions E2E tests not yet implemented")
+}
+
+// ---------------------------------------------------------------------------
+// REMOTE TESTS
+// Tests related to invoking processes remotely (in-cluster).
+// All remote tests presume the cluster has Tekton installed.
+// ---------------------------------------------------------------------------
+
+// TestRemote_Deploy ensures that the default action of running a remote
+// build succeeds:  uploading local souce code to the cluster for build and
+// delpoy in-cluster.
+//
+//	func deploy --remote
+func TestRemote_Deploy(t *testing.T) {
+	name := "func-e2e-test-remote-deploy"
+	_ = fromCleanEnv(t, name)
+
+	if err := newCmd(t, "init", "-l=go").Run(); err != nil {
+		t.Fatal(err)
+	}
+	if err := newCmd(t, "deploy", "--remote", "--builder=pack", "--registry=registry.default.svc.cluster.local:5000/func").Run(); err != nil {
```

Comment:
Registry is hardcoded here. Maybe a ClusterInternalRegistry var?

Reply:
This is now configurable